smoke/IotEdgeQuickstart/details/IotedgedLinux.cs

```diff
@@ -200,159 +222,296 @@ public Task Install()
             agentImage.ForEach(
                 image =>
                 {
-                    Console.WriteLine($"Setting up iotedged with agent image {image}");
+                    Console.WriteLine($"Setting up aziot-edged with agent image {image}");
                 },
                 () =>
                 {
-                    Console.WriteLine("Setting up iotedged with agent image 1.0");
+                    Console.WriteLine("Setting up aziot-edged with agent image 1.0");
                 });
 
-            const string YamlPath = "/etc/iotedge/config.yaml";
-            Task<string> text = File.ReadAllTextAsync(YamlPath);
-            var doc = new YamlDocument(await text);
+            const string KEYD = "/etc/aziot/keyd/config.toml";
+            const string CERTD = "/etc/aziot/certd/config.toml";
+            const string IDENTITYD = "/etc/aziot/identityd/config.toml";
+            const string EDGED = "/etc/aziot/edged/config.yaml";
+
+            Dictionary<string, (string owner, IConfigDocument document)> config = new Dictionary<string, (string, IConfigDocument)>();
+            config.Add(KEYD, ("aziotks", InitDocument(KEYD + ".default", true)));
+            config.Add(CERTD, ("aziotcs", InitDocument(CERTD + ".default", true)));
+            config.Add(IDENTITYD, ("aziotid", InitDocument(IDENTITYD + ".default", true)));
+            config.Add(EDGED, ("iotedge", InitDocument(EDGED + ".template", false)));
 
             method.ManualConnectionString.Match(
                 cs =>
                 {
-                    doc.ReplaceOrAdd("provisioning.device_connection_string", cs);
+                    string keyPath = "/var/secrets/aziot/keyd/device-id";
+                    config[IDENTITYD].document.ReplaceOrAdd("provisioning.source", "manual");
+                    config[IDENTITYD].document.ReplaceOrAdd("provisioning.authentication.method", "sas");
+                    config[IDENTITYD].document.ReplaceOrAdd("provisioning.authentication.device_id_pk", "device-id");
+                    config[KEYD].document.ReplaceOrAdd("preloaded_keys.device-id", $"file://{keyPath}");
+
+                    string[] segments = cs.Split(";");
```

Comment:
I'm not sure if this is a good idea or not. But maybe you can provide an example string of the current `cs` format?

Reply:
The format is always `HostName=X;DeviceId=Y;SharedAccessKey=Z`. If for some reason `cs` is not in this format, I expect one of the next lines to throw IndexOutofBoundsException.